nimare/io.py

```diff
@@ -275,3 +278,123 @@ def convert_sleuth_to_dataset(text_file, target="ale_2mm"):
         )
     dict_ = convert_sleuth_to_dict(text_file)
     return Dataset(dict_, target=target)
+
+
+def convert_neurovault_to_dataset(
+    collection_ids, contrasts, img_dir=None, map_type_conversion=None, **dset_kwargs
+):
+    """
+    Convert a group neurovault collections into a NiMARE Dataset.
+
+    Parameters
+    ----------
+    collection_ids : :obj:`list` of :obj:`int` or :obj:`dict`
+        A list of collections on neurovault specified by their id.
+        The collection ids can accessed through the neurovault API
+        (i.e., https://neurovault.org/api/collections) or
+        their main website (i.e., https://neurovault.org/collections).
+        For example, in this URL https://neurovault.org/collections/8836/,
+        `8836` is the collection id.
+        collection_ids can also be a dictionary whose keys are the informative
+        study name and the values are collection ids to give the collections
+        more informative names in the dataset.
+    contrasts : :obj:`dict`
+        Dictionary whose keys represent the name of the contrast in
+        the dataset and whose values represent how that contrast is identified
+        in neurovault.
+        For example, under the ``Name`` column in this URL
+        https://neurovault.org/collections/8836/,
+        a valid contrast could be "as-Animal", which will be called "animal" in the created
+        dataset if the contrasts argument is ``{'animal': ["as-Animal"]}``.
+    img_dir : :obj:`str` or None
+        Base path to save all the downloaded images, by default the images
+        will be saved to a temporary directory with the prefix "neurovault"
+    map_type_conversion : :obj:`dict` or None
+        Dictionary whose keys are what you expect the `map_type` name to
+        be in neurovault and the values are the name of the respective
+        statistic map in a nimare dataset. Default = None.
+    **dset_kwargs : keyword arguments passed to Dataset
+        Keyword arguments to pass in when creating the Dataset object.
+        see :obj:`nimare.dataset.Dataset` for details.
+
+    Returns
+    -------
+    :obj:`nimare.dataset.Dataset`
+        Dataset object containing experiment information from neurovault.
+    """
+    if img_dir is None:
+        img_dir = Path(tempfile.mkdtemp(prefix="neurovault_"))
+    else:
+        img_dir = Path(img_dir, exist_ok=True)
+        img_dir.mkdir(parents=True, exist_ok=True)
+
+    if map_type_conversion is None:
+        map_type_conversion = {
```

Comment:
Probably worth adding all of the other controlled options in NeuroVault that have an obvious mapping to NiMARE. Minimally, that's "Z map" and "p map"; not sure about others.

Reply:
Also, we might want to move this dictionary out of here and into its own module-level constant (e.g., `DEFAULT_MAP_TYPE_CONVERSION`), as users may want to add to it and pass that rather than having to specify all values from scratch.

Reply:
Created the constant and added "Z map" and "p map" keys.